### PR TITLE
Updates rln-chat2 interface

### PR DESCRIPTION
**Pull request by staheri14**
This PR is in line with https://github.com/status-im/nim-waku/issues/830 and comprises of the following updates:

- The `mountRlnRelay` procedure now accepts an optional `SpamHandler` proc to process the spam messages identified in the underlying installed pubsub topic validator. This option is used in chat2 to display the incoming spam messages to the client while marking them as spam. This optional spam handler enables us to illustrate the spam protection process to the chat2 users in a real-time fashion (otherwise, spam messages will be discarded silently in the relay protocol or the log file should be inspected to check the discarded spam messages). This feature is going to be used for the rln-relay testnet.

- A few utility procs are also introduced i.e., `getChatLine` and `toRLNSignal`.

### CODE REVIEW

File: examples/v2/chat2.nim [hunk 1]
```diff
@@ -116,7 +116,6 @@ proc generateSymKey(contentTopic: ContentTopic): SymKey =
   symKey
 
 proc connectToNodes(c: Chat, nodes: seq[string]) {.async.} =
-  echo "Connecting to nodes"
```
Comment (oskarth) [inline examples/v2/chat2.nim:119]: Maybe leave as `log`?

File: examples/v2/chat2.nim [hunk 2]
```diff
@@ -225,6 +250,7 @@ proc publish(c: Chat, line: string) =
             debug "could not append rate limit proof to the message", success=success
           else:
             debug "rate limit proof is appended to the message", success=success
+            echo "--rln epoch: ", fromEpoch(message.proof.epoch)
```
Comment (oskarth) [inline examples/v2/chat2.nim:253]: For dogfooding we can leave it as echo, we might want to move it to log later though

**Review by oskarth — APPROVED** [review]
LGTM! What are next steps for dogfooding?

**Review by jm-clius — APPROVED** [review]
LGTM. Perhaps we can create a new default content topic for RLN dogfooding? We'll just need a good name :)

**Comment by staheri14** [comment]
> LGTM. Perhaps we can create a new default content topic for RLN dogfooding? We'll just need a good name :)

@jm-clius The name is already picked by Oskar: `Luzhou`

**Comment by staheri14** [comment]
> LGTM. Perhaps we can create a new default content topic for RLN dogfooding? We'll just need a good name :)

I assume we do not want to touch the default `contentTopic` of chat2 (the one in the [`config_chat2` file](https://github.com/status-im/nim-waku/blob/5a890b5f90d877fa45c2a5b0f33a5b3b777da831/examples/v2/config_chat2.nim#L211))? as it might be still in use and changing the default may be a breaking change. One immediate example changing the default will affect the chat2 tutorial.
Thus, for the rln-relay dogfooding, what I think is that we shall pass the rln-relay content topic i.e., `/toy-chat/2/Luzhou/proto` when running chat2 instances locally. WDYT @jm-clius?

File: examples/v2/chat2.nim (hunk 1 above)
Comment (staheri14) [reply examples/v2/chat2.nim]: sure, done

**Comment by staheri14** [comment]
> What are next steps for dogfooding?

**Comment by oskarth** [comment]
Nitpick: let's make it lowercase for consistency `/toy-chat/2/Luzhou/proto`

**Comment by staheri14** [comment]
> Nitpick: let's make it lowercase for consistency `/toy-chat/2/Luzhou/proto`

sure, done https://github.com/status-im/nim-waku/pull/846/commits/ec5da50a299e68ab1fa4b4e3869e49187554685c

**Pull request merged**